### PR TITLE
use_existing_tensors flag

### PR DESCRIPTION
**Pull request by pandu-k**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature: use_existing_tensors flag

* **What is the current behavior?** (You can also link to an open issue here)
No way to re-use existing tensor facets

* **What is the new behavior (if this is a feature change)?**
Use existing tensor facets during add documents by setting use_existing_tensors=True

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
Yes

* **Related Python client changes** (link commit/PR here)
https://github.com/marqo-ai/py-marqo/pull/59

* **Related documentation changes** (link commit/PR here)

* **Other information**:
There are two GET requests per doc (all of these are batched into a single request to Marqo-OS). But this means we are not atomic in the relationship between vectors and the document content. We may be able to solve this via optimistic locking, by asserting each doc's version must be the same 

* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Comment by pandu-k** [comment]
Unit tests running: https://github.com/marqo-ai/marqo/actions/runs/4214855171

**Comment by pandu-k** [comment]
Py-Marqo PR: https://github.com/marqo-ai/py-marqo/pull/59

**Comment by pandu-k** [comment]
Related closed PR: https://github.com/marqo-ai/marqo/pull/334

File: tests/tensor_search/test_add_documents.py [hunk 1]
```diff
@@ -929,7 +929,7 @@ def test_put_documents_orchestrator(self):
                   {"_id": "789", "Temp": 12.5},
                   ],
             auto_refresh=True, update_mode='update', processes=4, batch_size=1)
-        time.sleep(3)
+        time.sleep(5)
```
Comment (vicilliar) [inline tests/tensor_search/test_add_documents.py:932]: what was the purpose of increasing this time?

**Comment by vicilliar** [comment]
passed unit tests. test variable names updated. ready for merge.

**Review by vicilliar — APPROVED** [review]
passed unit tests. test variable names updated. ready for merge.

**Pull request merged**